mage_integrations/mage_integrations/destinations/kafka/__init__.py

```diff
@@ -0,0 +1,66 @@
+import argparse
+import ast
+import json
+import sys
+from typing import Dict, List
+
+from kafka import KafkaConsumer, KafkaProducer
+
+from mage_integrations.destinations.base import Destination
+from mage_integrations.destinations.constants import KEY_RECORD
+from mage_integrations.destinations.utils import update_record_with_internal_columns
+
+MAXIMUM_BATCH_SIZE_MB = 1
+
+
+class Kafka(Destination):
+
+    def test_connection(self) -> None:
+        consumer = KafkaConsumer(bootstrap_servers=self.config['bootstrap_server'],
+                                 api_version=ast.literal_eval(self.config.get('api_version',
+                                                                              '(0, 10, 2)')))
+        topics = consumer.topics()
+
+        if not topics:
+            consumer.close()
+            raise Exception('Kafka was not able to connect to BootStrap Server')
+        elif topics:
+            consumer.close()
+            return True
+
+    def build_client(self):
+        kwargs = dict(
+            bootstrap_servers=self.config['bootstrap_server'],
+            api_version=ast.literal_eval(self.config.get('api_version', '(0, 10, 2)')),
+            value_serializer=lambda x: json.dumps(x).encode('utf-8'),
+            key_serializer=lambda x: x.encode('utf-8') if x else None,
+        )
+        producer = KafkaProducer(**kwargs)
+        return producer
+
+    def export_batch_data(self, record_data: List[Dict], stream: str, tags: Dict = None) -> None:
+
+        self.logger.info('Export data started.')
+
+        producer = self.build_client()
+
+        if self.key_properties is not None:
+            key_properties = self.key_properties[stream][0]
```

Comment:
nit: let's make sure `key_properties[stream]` is not None and has more than one item. Otherwise, it'll throw exception.

Reply:
Modified in [477624f](https://github.com/mage-ai/mage-ai/pull/3976/commits/477624ff40d9a1330d77ce4ff20eb11be06f3efd)